design-system/src/theme-provider.tsx

```diff
@@ -92,19 +92,47 @@ ThemeProvider.defaultProps = {
   theme: 'default',
 };
 
-const useTheme = (parentSelector = defaultParentSelector): ThemeName => {
+type TUseThemeResult = {
+  currentTheme: ThemeName;
+};
+const useTheme = (parentSelector = defaultParentSelector): TUseThemeResult => {
   const [theme, setTheme] = useState<ThemeName>('default');
   const parentSelectorRef = useRef(parentSelector);
+  const observerRef = useRef(
+    new MutationObserver((mutationList, _observer) => {
```

Comment:
I think this is a great suggestion but I would like to do it in a follow-up PR since I'll be introducing something new and I think the code from app-kit should be updated. I expect some discussion around it so I think it's better to do it as a follow-up.

Reply:
@emmenko @nkuehn and @CarlosCortizasCT 
Upgrading `docs` repo here, and I'm seeing gatsby SSR failing because of the use of `MutationObserver`.
Any chance we can get @emmenko suggestion implemented soon?

Reply:
Hi @gabriele-ct 

We have [some work in progress](https://github.com/commercetools/ui-kit/pull/2351) but its priority has not been big enough since other MC redesign tasks need more attention.
We will try to find some time to finish it.

Just for us to know, is [this workaround](https://github.com/commercetools/commercetools-docs-kit/pull/1418) not working anymore?